backend/routes/rankings.js

```diff
@@ -0,0 +1,138 @@
+const express = require("express");
+
+// allows routes to be sent out
+const router = express.Router();
+const db = require("../mongo.js");
+
+const { isAuthenticated } = require("../middlewares/auth");
+
+router.use(isAuthenticated);
+
+router.get("/:userid/:programId", function(req, res) {
+  db["Authentication"].rankings
+    .aggregate([
+      {
+        $match: { userId: req.params.userid, programId: req.params.programId }
+      },
+      {
+        $unwind: "$applications"
+      },
+      {
+        $lookup: {
+          from: "Reviews",
+          let: {
+            appId: "$applications.appId",
+            userId: "$userId",
+            programId: "$programId"
+          },
+          pipeline: [
+            {
+              $match: {
+                $expr: {
+                  $and: [
+                    { $eq: ["$applicationId", "$$appId"] },
+                    { $eq: ["$userId", "$$userId"] },
+                    { $eq: ["$programId", "$$programId"] }
+                  ]
+                }
+              }
+            },
+            {
+              $addFields: {
+                questionRatings: {
+                  $map: {
+                    input: "$sectionList",
+                    as: "item",
+                    in: {
+                      $cond: {
+                        if: { $gte: ["$$item.rating", 0] },
+                        then: "$$item.rating",
+                        else: null
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            {
+              $project: {
+                rating: 1,
+                suggested: {
+                  $avg: "$questionRatings"
+                }
+              }
+            }
+          ],
+          as: "ratingInfo"
+        }
+      },
+      {
+        $lookup: {
+          from: "Applications",
+          let: { appId: "$applications.appId" },
+          pipeline: [
+            { $match: { $expr: { $eq: ["$_id", "$$appId"] } } },
+            {
+              $project: {
+                "Organization Name": 1,
+                "Organization Name (legal name)": 1
```

Comment:
are these meant to be the identifier? users can rename this field so this may not always be the org name

Reply:
Yes. Good catch